kratos/containers/sparse_graph.h

```diff
@@ -0,0 +1,394 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+
+#if !defined(KRATOS_SPARSE_GRAPH_H_INCLUDED )
+#define  KRATOS_SPARSE_GRAPH_H_INCLUDED
+
+
+// System includes
+#include <iostream>
+#include "includes/ublas_interface.h"
+#include "includes/serializer.h"
+
+// External includes
+#include <unordered_map>
+#include <unordered_set>
+
+// Project includes
+#include "includes/define.h"
+
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+//class to construct and store a matrix graph. Can be used to construct efficiently a CSR matrix (or other sparse matrix types)
+
+/** This class is designed to store a matrix graph, aimed at the fast construction of other
+ * sparse matrix formats (particularly CSR)
+ * IMPORTANT NOTE: it is BY DESIGN NOT threadsafe! (a graph should be computed in each thread and then merged)
+*/
+
+class SparseGraph
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    typedef std::size_t IndexType;
+    typedef std::map<IndexType, std::unordered_set<IndexType> > GraphType; //using a map since we need it ordered
```

Comment:
Is it a map, and not a vector, to be usable in the distributed version or is it for sank of generality?